python/yugabyte_db_thirdparty/yb_build_thirdparty_main.py

```diff
@@ -93,6 +94,23 @@ def main() -> None:
         log("Time taken for packaging/upload %.1f sec",
             time.time() - packaging_and_upload_start_time_sec)
 
+    if builder.args.snyk:
+        snyk_token = os.environ.get('SNYK_TOKEN')
+        if snyk_token is None:
+            log("SNYK_TOKEN is not set, not running snyk.")
+        else:
+            log("Running Snyk Vulnerability Scan.")
+            os_type = os.environ.get('OSTYPE', '')
```

Comment:
We already have utility functions for OS detection (Linux vs macOS), used widely within this codebase.